### PR TITLE
[Merged by Bors] - feat: add some ring theory lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
From flt-regular.
    
Co-authored-by: Andrew Yang <the.erd.one@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Ideal/Norm.lean [hunk 1]
```diff
@@ -441,6 +465,13 @@ theorem finite_setOf_absNorm_eq [CharZero S] {n : ℕ} (hn : 0 < n) :
     congr
 #align ideal.finite_set_of_abs_norm_eq Ideal.finite_setOf_absNorm_eq
 
+theorem norm_dvd_iff (x : S) (hx : Prime (Algebra.norm ℤ x)) {y : ℤ} :
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Ideal/Norm.lean:468]:
```suggestion
theorem norm_dvd_iff {x : S} (hx : Prime (Algebra.norm ℤ x)) {y : ℤ} :
```

File: Mathlib/RingTheory/DedekindDomain/Ideal.lean [hunk 2]
```diff
@@ -875,6 +875,9 @@ theorem inf_eq_mul_of_coprime {I J : Ideal A} (coprime : I ⊔ J = ⊤) : I ⊓
   rw [← associated_iff_eq.mp (gcd_mul_lcm I J), lcm_eq_inf I J, gcd_eq_sup, coprime, top_mul]
 #align ideal.inf_eq_mul_of_coprime Ideal.inf_eq_mul_of_coprime
 
+theorem isCoprime_iff_gcd {I J : Ideal A} : IsCoprime I J ↔ gcd I J = 1 := by
```
[the remark this answers is not in the document]
Comment (Ruben-VandeVelde) [reply Mathlib/RingTheory/DedekindDomain/Ideal.lean:878]: I added `Ideal.isCoprime_iff_add`; is that what you were thinking about?

[the remark this answers is not in the document]
Comment (riccardobrasca) [reply Mathlib/RingTheory/DedekindDomain/Ideal.lean:878]: Yes, thanks.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

File: Mathlib/RingTheory/Ideal/Operations.lean [hunk 3]
```diff
@@ -825,6 +825,29 @@ theorem span_pair_mul_span_pair (w x y z : R) :
   simp_rw [span_insert, sup_mul, mul_sup, span_singleton_mul_span_singleton, sup_assoc]
 #align ideal.span_pair_mul_span_pair Ideal.span_pair_mul_span_pair
 
+theorem isCoprime_iff_sup : IsCoprime I J ↔ I ⊔ J = ⊤ := by
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Operations.lean:828]:
```suggestion
theorem isCoprime_iff_codisjoint : IsCoprime I J ↔ Codisjoint I J := by
```

File: Mathlib/RingTheory/Subsemiring/Basic.lean [hunk 4]
```diff
@@ -168,6 +168,9 @@ instance toLinearOrderedCommSemiring {R} [LinearOrderedCommSemiring R] [SetLike
     (fun _ _ => rfl) (fun _ _ => rfl) (fun _ => rfl) (fun _ _ => rfl) fun _ _ => rfl
 #align subsemiring_class.to_linear_ordered_comm_semiring SubsemiringClass.toLinearOrderedCommSemiring
 
+instance toCharZero [CharZero R] : CharZero s :=
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Subsemiring/Basic.lean:171]:
```suggestion
instance instCharZero [CharZero R] : CharZero s :=
```

**Comment by riccardobrasca** [comment]
I see Eric has some comments.

bors r-

**Comment by riccardobrasca** [comment]
bors d+

**Pull request closed without merging**